src/rust_toolchain.rs

```diff
@@ -9,15 +9,46 @@ pub fn create() {
 
 #[cfg(test)]
 mod tests {
+    use std::env;
+
     use reqwest::blocking;
 
     use super::CONTENTS;
 
+    const TEST_BRANCH_NAME: &str = "dev";
+    const CRON_JOB_BRANCH_NAME_ENV_VAR: &str = "BRANCH_SELECTOR";
+    const PR_TARGET_BRANCH_NAME_ENV_VAR: &str = "GITHUB_BASE_REF";
+    const CI_BRANCH_NAME_ENV_VAR: &str = "GITHUB_REF_NAME";
     const CASPER_NODE_TOOLCHAIN_URL: &str =
         "https://raw.githubusercontent.com/casper-network/casper-node/dev/smart_contracts/rust-toolchain";
 
+    /// Checks that the pinned version of Rust is that of the `dev` branch of casper-node.
+    ///
+    /// If none of `BRANCH_SELECTOR`, `GITHUB_BASE_REF` and `GITHUB_REF_NAME` are set, or they're
+    /// all set to empty strings, the test is an auto-pass.
+    ///
+    /// For testing locally, you can manually run e.g.
+    /// ```
+    /// BRANCH_SELECTOR=dev cargo t
+    /// ```
     #[test]
-    fn check_toolchain_version() {
+    fn check_toolchain_version_on_dev() {
+        if let Ok(true) = env::var(CRON_JOB_BRANCH_NAME_ENV_VAR)
+            .or_else(|_| env::var(PR_TARGET_BRANCH_NAME_ENV_VAR))
+            .or_else(|_| env::var(CI_BRANCH_NAME_ENV_VAR))
```

Comment:
Is there a reason behind the ordering or is it random? Also, should we ensure that if 2 or more of those variables exist, they should be the same (i.e. `"dev"`)?

Reply:
The ordering is random, but I expect `BRANCH_SELECTOR` to be set for scheduled CI runs, `GITHUB_BASE_REF` for PRs and `GITHUB_REF_NAME` for any other CI runs (at least according to the comments I added to these env vars in the integration test file).  If more than one exists, I'd expect them to have the same value, but I think it's probably overkill to actually check for that.